### DISCUSSION TITLE
Password protected services in v4

### DISCUSSION THREAD

**Discussion by deployn** [coollabsio/coolify#1269] [Feature Requests]
Hey,

it was possible to protect a service with a basic user:password credential in version 3.
In version 4 I can't find this option.

**Comment by drubb** [comment]
Definitely a valuable feature, I'm missing this too.

**Comment by Rechdan** [comment]
I have a docker compose project and a setting for that would be awesome, specially for a `dev` environment.

**Comment by drubb** [comment]
Seems there is support for Basic Auth in V4 using custom container labels:
https://coolify.io/docs/traefik/basic-auth
Tried adding them to my docker-compose file, but no success for now:
```
labels:
      - 'traefik.http.middlewares.custom-auth.basicauth.users=test:$$apr1$$9pt2edko$$Tuk6M975nbBzM3Mtp1Fdb/'
      - traefik.http.routers.https-0-wwc48wo.middlewares=custom-auth
```
Especially I'm not sure how the container id in the second label (wwwc48wo above) should be set, for now I took the one visible in the deployable compose file for other traefik middlewares.
But I guess there's still a missing piece.

**Reply by marko-hologram** [reply]
Did you manage to resolve this? I don't use Traefik, but use Caddy and managed to get this working quite easily:

```
caddy_0.basicauth.0_testuser=$2a$14$adia0TeEZnVY5aWsdyZ2AeMNwMBBC.Jzv3r3qr.nRs/aQFHv9Vhti
```

I'm not sure that you need to wrap your labels in single quotes at all. I know I didn't have to since this is copy pasted from my Coolify instance where I'm testing this.

**Reply by stevenirby** [reply]
> ```
> caddy_0.basicauth.0_testuser=$2a$14$adia0TeEZnVY5aWsdyZ2AeMNwMBBC.Jzv3r3qr.nRs/aQFHv9Vhti
> ```

Thank you so much!

**Comment by marko-hologram** [comment]
Yeah I would love this for main deploy and for preview deployments as well. If I'm working on a feature in a pull request, I don't want someone random to be able to open the project using preview deployment link. I want all links for my project to be protected.

This is currently possible with custom labels on main deployment, but I haven't seen a way to do it for preview deployments as well. Although if it could be set for main deployment with a dedicated UI for it instead of using reverse proxy labels, that would make it a bit friendlier to use.

**Comment by IwishIcanFLighT** [comment]
Is there any plan for this feature? I'm using the middlewares basic-auth approach to mitigate my need, but it keeps getting reset/removed every time I change/save my service's settings in the dashboard... 🙄

**Reply by marko-hologram** [reply]
If I understand you correctly, you need to set labels to readonly so that Coolify doesn't modify them.

![image](https://github.com/user-attachments/assets/37ff82bb-7feb-4b82-adb3-838dc5589214)

I've erased my labels because I don't want to expose anything here, but after I've set labels to readonly, I can update them and save and Coolify won't override anything.

**Reply by IwishIcanFLighT** [reply]
That worked, thanks!

**Comment by benjaminstary** [comment]
in [v4.0.0-beta.434](https://github.com/coollabsio/coolify/releases/tag/v4.0.0-beta.434) it is pressent in the UI (General page)
<img width="794" height="163" alt="image" src="https://github.com/user-attachments/assets/7f056693-926d-42ea-8cd0-a32960cdae05" />

**Reply by LarsNorgaard** [reply]
I can't seem to find this. Could you guide us through?

**Reply by julianschnell** [reply]
I also don't see this feature (v4.0.0-beta.446) - has basic auth been removed again or is this only activated for certain resources?

**Reply by benjaminstary** [reply]
Hm interesting, all of my private github repos has it automatically present under `Configuration -> General -> HTTP Basic Authentication`

**Reply by mark-antal-csizmadia** [reply]
This works for me, I'm on `v4.0.0-beta.442`. Great feature!
